### PR TITLE
Add test for tokens with a list of audiences

### PR DESCRIPTION
**Pull request by lukasz-walkiewicz**

### CODE REVIEW

File: core/trino-main/src/test/java/io/trino/server/security/oauth2/TestOAuth2WebUiAuthenticationFilter.java [hunk 1]
```diff
@@ -108,27 +111,28 @@ public TestOAuth2WebUiAuthenticationFilter()
     public void setup()
             throws Exception
     {
-        Logging.initialize();
+        Logging logging = Logging.initialize();
+        logging.setLevel(OAuth2WebUiAuthenticationFilter.class.getName(), Level.DEBUG);
```
Comment (kokosing) [inline core/trino-main/src/test/java/io/trino/server/security/oauth2/TestOAuth2WebUiAuthenticationFilter.java:115]: undo

**Review by kokosing — COMMENTED** [review]
Ping me once it is ready to review.

File: core/trino-main/src/test/java/io/trino/server/security/oauth2/TestingHydraIdentityProvider.java [hunk 2]
```diff
@@ -13,76 +13,103 @@
...
     }
 
     public void start()
+            throws Exception
     {
-        consentContainer.start();
+        loginAndConsentServer = createTestingLoginAndConsentServer();
+        closer.register(loginAndConsentServer::stop);
+        loginAndConsentServer.start();
+        URI loginAndConsentBaseUrl = loginAndConsentServer.getBaseUrl();
```
Comment (kokosing) [inline core/trino-main/src/test/java/io/trino/server/security/oauth2/TestingHydraIdentityProvider.java:88]: How does it work? It looks like you are accessing host port from docker. Is it just because http server is not bound on localhost?

**Review by ksobolew — APPROVED** [review]
Looks good to me overall

File: core/trino-main/src/test/java/io/trino/server/security/oauth2/TestingHydraIdentityProvider.java (hunk 2 above)
Comment (lukasz-walkiewicz) [reply core/trino-main/src/test/java/io/trino/server/security/oauth2/TestingHydraIdentityProvider.java:88]: There's no communication hydra -> login & consent service, only login & consent service -> hydra, so it's not a problem that hydra cannot access a service on the host.
That said, hydra needs login & consent urls configured because it redirects the client to those urls to log in and/or give consent so the client needs access.

File: core/trino-main/src/test/java/io/trino/server/security/oauth2/TestOAuth2WebUiAuthenticationFilter.java [hunk 3]
```diff
@@ -108,27 +111,28 @@ public TestOAuth2WebUiAuthenticationFilter()
...
                 CLIENT_SECRET_POST,
-                TRINO_AUDIENCE,
+                ImmutableList.of(TRINO_AUDIENCE),
                 EXPOSED_SERVER_URL + "/oauth2/callback");
         hydraIdP.createClient(
                 UNTRUSTED_CLIENT_ID,
                 UNTRUSTED_CLIENT_SECRET,
                 CLIENT_SECRET_POST,
-                UNTRUSTED_CLIENT_AUDIENCE,
+                ImmutableList.of(UNTRUSTED_CLIENT_AUDIENCE),
```
[the remark this answers is not in the document]
Comment (ksobolew) [reply core/trino-main/src/test/java/io/trino/server/security/oauth2/TestOAuth2WebUiAuthenticationFilter.java]: Let me rephrase a bit: *would* it make sense to add such a test? :)

[the remark this answers is not in the document]
Comment (lukasz-walkiewicz) [reply core/trino-main/src/test/java/io/trino/server/security/oauth2/TestOAuth2WebUiAuthenticationFilter.java]: I don't think that's possible. We can get an access token only for audience(s) registered for the client.

**Review by kokosing — APPROVED** [review]

**Comment by kokosing** [comment]
Merged, thanks!
